### PR TITLE
Sort portfolio rows and investments

### LINKED ISSUE #700: Sort Portfolio Table by importance
Unsurprisingly, allowing users to drag their categories around is difficult and not worth our time right now.

But, we can still order the portfolio table in terms of importance.

Rows will be sorted by percentile, investments in each row will be sorted by percent return

### PR DESCRIPTION
**Pull request by ritmatter**
Fixes #700

### CODE REVIEW

File: js/components/PortfolioTable.jsx [hunk 1]
```diff
@@ -39,6 +52,7 @@ var PortfolioTable = React.createClass({
         totalDividends += category.investments[j].dividend;
       }
     }
+    totalDividends = Math.floor(totalDividends * 100)/100;
```
Comment (ritmatter) [inline js/components/PortfolioTable.jsx]: This fixes a strange error where sometimes adding a sum of numbers together created a really long number.

File: js/components/InvestmentList.jsx [hunk 2]
```diff
@@ -4,11 +4,31 @@ var InvestmentItem = require('./InvestmentItem');
 var React = require('react');
 
 var InvestmentList = React.createClass({
+  getOrderedInvestments: function() {
+    var comparator = this.getInvestmentComparator();
```
Comment (krazemon) [inline js/components/InvestmentList.jsx]: Just making sure we're on the same page, `sort()` will also change the value of `this.props.investments`.

File: js/components/PortfolioTable.jsx [hunk 3]
```diff
@@ -16,9 +16,10 @@ var PortfolioTable = React.createClass({
 
   getPortfolioItems: function(privateFields) {
     var portfolioItems = [];
-    var length = this.props.user.portfolio.length;
-    for (var i = 0; i < length; i++) {
-      var category = this.props.user.portfolio[i];
+    var portfolio = this.props.user.portfolio;
+    var sortedPortfolio = this.props.user.portfolio.sort(this.getPortfolioComparator());
```
Comment (krazemon) [inline js/components/PortfolioTable.jsx]: Same thing here. Calling sort on `this.props.user.portfolio` will sort in place and return the new Array.

**Comment by krazemon** [comment]
Address my comments and I'll take a look. The things I'm pointing out aren't necessarily blocking but I would like them commented upon, as weird stuff might happen when you try to modify props.

**Pull request closed without merging**